uiowa/uiowa
# Transfer field_teaser to summary fields

Resolves #6674 





# How to test


```
ddev blt frontend
```

```
ddev blt ds --site=default && ddev drush @default.local uli
```
```
ddev blt ds --site=admissions.uiowa.edu && ddev drush @admissions.local uli
```
Check for clean syncs

For all default content types (article, person, page):

- When editing existing content, the body summary field should be populated with content (except for the few cases where the teaser had not originally been filled out)
- When adding new content, the body summary field should be visible and usable
- The `field_teaser` field should be present in its usual location, but disabled and with help text to direct you to the summary field
- The summary field, even if it differs from the old `field_teaser` value, should be what gets used in teaser displays (article/people lists, featured content, etc)
- The summary field should be used for the node's meta description (if that description has not been overridden in the Advanced SEO settings) for both new and existing content
- If the summary field is empty, a trimmed chunk of the body should be used for the node's meta description (if that description has not been overridden in the Advanced SEO settings) for both new and existing content
- If an existing list block had been hiding the description, it should continue to hide the description

For People

- Both default and teaser displays should not have visually changed from prod
- The edit form should appear the same as in prod/main, with the exception that the summary field is now included with the Biography field, and the short bio field is disabled (with text pointing to the summary field)
- The Biography field should be populated with the same text and format as in prod
- A codebase search for `field_person_bio` or `field-person-bio` should only result in references to `field_person_bio_headline`/`field-person-bio-headline` or migration-related code

For Articles

```
ddev blt ds --site=now.uiowa.edu && ddev drush @now.local uli
```

- Check editing an article, and that the form matches prod (with the addition of the Summary field being visible, usable)
- Check for other splits of Articles that may need to be accounted for, particularly complete splits of the form or teaser displays

For Issues
```
ddev drush @default.local config-split:activate periodical
```
Alternatively, 
```
ddev blt ds --site=91stmeridian.iwp.uiowa.edu && ddev drush @iwp91stmeridian.local uli
```
- Run through the "for default content types" guidelines for the Issue content type


Residence halls
```
ddev blt ds --site=housing.uiowa.edu && ddev drush @housing.local uli
```

- Run through the "for default content types" guidelines for the Residence Hall content type

## V2

```
ddev blt ds --site=hawkeyemarchingband.uiowa.edu && ddev drush @hawkeyemarchingband.local uli
```

- Check that you can still access, edit `field_teaser` for Page content type, and that it is used for meta description and teaser displays
- Check that the updates are in place for using the summary in place of the teaser for articles, people, and that they are properly used in the teaser displays/descriptions

## Default install

Check for a successful Percy build

```
ddev blt di --site=default && ddev drush @default.local uli /admin/content
```

Check the default content and see that they utilize the `body:summary` field rather than `field_teaser`
Check that the default person content 

## Config updates

Check for clean syncs in the following:

- Features
  - event
  - grad_person_extended
  - periodical
 - Sites
   - admissions.uiowa.edu
   - housing.uiowa.edu
   - immuno.grad.uiowa.edu
   - isa.uiowa.edu
   - now.uiowa.edu
   - uiowa.edu
   - uipress.uiowa.edu 

## 9/7 Update:
To fix:
- [x] v2 articles - field_teaser not transferring to summary field (and so not used in meta description)
- [x] v2 pages 
   - [x] no meta description
   - [x] no teaser display

Note: I (Marie) have been testing v2 on mnh.uiowa.edu as I found more content using the field-teaser.

To Dos from others' lists I have not gotten to:
- [x] Closer inspection of update hooks
- [x] Update hook spot checking
- [x] Inspect oddities found in person token, isa person form display
- [x] Check for other splits of Articles that may need to be accounted for, particularly complete splits of the form or teaser displays

## additional testing based on recent changes:
- Sync v3 site, edit existing or create a featured content block -  sandbox has existing blocks at https://sandbox.uiowa.ddev.site/featured-content. Test that the 'Hide Descriptions' style works as expected.
- Review update hook code changes. To check results:
  - Sync mnh.uiowa.edu, check that existing v2 articles have summary field filled in (for part of update hook handling null summary fields)
  - Sync hawkeyemarchingband.uiowa.edu, check that existing v2 articles have summary field filled in (for part of update hook handling nodes with no body content)
- On one of the already synced v2 sites or a new one:
  - Inspect a few page nodes to check the meta description is coming from the field_teaser. 
  - In a featured content block, check that pages have a teaser display
  - In a featured content, articles, or people block, check 'Hide Summaries' style works as expected.
